core/ledger/pvtdatastorage/reconcile_missing_pvtdata.go

```diff
@@ -21,91 +22,169 @@ import (
 //     from the above created data entries
 // (2) create a db update batch from the update entries
 // (3) commit the update batch to the pvtStore
-func (s *Store) CommitPvtDataOfOldBlocks(blocksPvtData map[uint64][]*ledger.TxPvtData) error {
+func (s *Store) CommitPvtDataOfOldBlocks(
+	blocksPvtData map[uint64][]*ledger.TxPvtData,
+	deprioritizedMissingData ledger.MissingPvtDataInfo,
+) error {
+	s.purgerLock.Lock()
+	defer s.purgerLock.Unlock()
+
```

Comment:
The main reason for holding this lock is to ensure that purgeMgr is not purging entries that are being added here. With complex scheduling, it is possible that data &  expiry entry is added to the in-memory batch but purger removed it. We still commit the data & expiry entry and in the next cycle of purging, the purge will remove the added entry. It is not really a bad consistent issue but keeping this lock seems better so that all data are removed at the expiring block height.